cypress/e2e/environment.cy.js

```diff
@@ -0,0 +1,389 @@
+    inputType: "select",
+    newValue: "push_incremental_deploy",
+  },
+  {
+    name: "lsm_partial_compile",
+    inputType: "switch",
+    newValue: "true",
+  },
+  {
+    name: "notification_retention",
+    inputType: "text",
+    newValue: "375",
+  },
+  {
+    name: "protected_environment",
+    inputType: "switch",
+    newValue: "true",
+  },
+  {
+    name: "purge_on_delete",
+    inputType: "switch",
+    newValue: "true",
+  },
+  {
+    name: "push_on_auto_deploy",
+    inputType: "switch",
+    newValue: "false",
+  },
+  {
+    name: "resource_action_logs_retention",
+    inputType: "text",
+    newValue: "8",
+  },
+  {
+    name: "server_compile",
+    inputType: "switch",
+    newValue: "false",
+  },
+];
+const newEnvCard = '[aria-label="Overview-Success"] > :first-child';
+const projectNameInput = '[aria-label="Project Name-typeahead"]';
+const newProjectName = '[aria-label="Project Name-select-input';
+const newProjectNameButton = ".pf-c-select__menu-item";
+const nameInput = '[aria-label="Name-input"]';
+const descInput = '[aria-label="Description-input"]';
+const repoInput = '[aria-label="Repository-input"]';
+const branchInput = '[aria-label="Branch-input"]';
+const iconInput = "#simple-text-file-filename";
+
+const testProjectName = (number) => "Test Project Name " + number;
+const testName = (number) => "TestName " + number;
+
+const createEnv = (number, shouldTypeName = true, isShort = false) => {
+  cy.get(projectNameInput).type(testProjectName(number));
+  cy.get(newProjectName).within(() => {
+    cy.get(newProjectNameButton).click();
+  });
+  if (shouldTypeName) {
+    cy.get(nameInput).type(testName(number));
+  }
+  if (!isShort) {
+    cy.get(descInput).type("Test description");
+    cy.get(repoInput).type("repository");
+    cy.get(branchInput).type("branch");
+    cy.get(iconInput).attachFile(icon);
+  }
+};
+
+const deleteEnv = (number) => {
+  cy.get("button").contains("Delete environment").click();
+  cy.get('[aria-label="delete"]').should("be.disabled");
+
+  cy.get('[aria-label="delete environment check"]').type(testName(number));
+  cy.get('[aria-label="delete"]').should("not.be.disabled").click();
+  cy.url().should("eq", Cypress.config().baseUrl + "/console/");
+  cy.get('[aria-label="Environment card"]')
+    .contains(testProjectName(number))
+    .should("not.exist");
+};
+
+describe("Environment", function () {
+  it("1.1 cancel creation of an environment", function () {
+    cy.visit("/console/");
+    cy.get(newEnvCard).click();
+    cy.url().should(
+      "eq",
+      Cypress.config().baseUrl + "/console/environment/create"
+    );
+    createEnv(1);
+    cy.get("button").contains("Cancel").click();
+    cy.url().should("eq", Cypress.config().baseUrl + "/console/");
+  });
+
+  it("1.2 Create ne environment", function () {
+    //fill the form and submit
+    cy.visit("/console/environment/create");
+    createEnv(2, false);
+    cy.get("button").contains("Submit").should("be.disabled");
```

Comment:
I strongly oppose this in this case.

There are reasons to factor out common code, but I don't see which ones apply here. 

For me, in testcases, the rule is that it should read as a story. It should, at every point, be clear what is happening.
At this point, this is very readable. It relates well to what is visible. 

By factoring out this trivial code, it becomes harder to read, because we replace something concrete with something abstract, that may not be as clear as the current code. 

At the same time, it reduces locality: to modify this assertion, the modification has to be done somewhere else. 
It also tangles the testcases: when we update the assert, we have to check that it does not affect any other place it is used. 

In general, for me, the idea is not to factor out common code, but strong abstractions. 
(I can explain this more in detail in a meeting)

Reply:
As discussed we will only extract reussable sections when they appear to be usefull in other test scenarios.